# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/new-regexp/regexp-shim.cc
# Commit: 18b93628199b
# Full Hash: 18b93628199b312a826b426162d0a43883b2edae
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-06 21:51:14
# Regressor Bug: 1634135
# File Overlap Count: 1
# Description:
#   Bug 1634135: Disable named capture parsing until fully supported r=mgaudet
#   
#   The engine supports parsing named captures, but we don't have the code in place to expose the captured groups. Until we do, this code will make sure that we get a syntax error when parsing them.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73115
# ==============================================================================

diff -r 6aaf6e87a5d1 -r 18b93628199b js/src/new-regexp/regexp-shim.cc
--- a/js/src/new-regexp/regexp-shim.cc	Tue May 05 12:49:53 2020 +0000
+++ b/js/src/new-regexp/regexp-shim.cc	Tue May 05 12:49:54 2020 +0000
@@ -183,7 +183,7 @@
 
 Handle<FixedArray> Isolate::NewFixedArray(int length) {
   MOZ_RELEASE_ASSERT(length >= 0);
-  MOZ_CRASH("TODO");
+  return Handle<FixedArray>::fromHandleValue(JS::UndefinedHandleValue);
 }
 
 template <typename CharT>
